resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -716,7 +732,92 @@ public int attempts() {
         }
 
         /**
-         * Get metricsof the CircuitBreaker
+         * Get metrics of the CircuitBreaker
+         */
+        @Override
+        public CircuitBreakerMetrics getMetrics() {
+            return circuitBreakerMetrics;
+        }
+    }
+
+    private class MetricsOnlyState implements CircuitBreakerState {
+
+        private final CircuitBreakerMetrics circuitBreakerMetrics;
+
+        MetricsOnlyState() {
+            this.circuitBreakerMetrics = CircuitBreakerMetrics
+                .forMetricsOnly(getCircuitBreakerConfig());
+        }
+
+        /**
+         * Returns always true, because the CircuitBreaker is always closed in this state.
+         *
+         * @return always true, because the CircuitBreaker is always closed in this state.
+         */
+        @Override
+        public boolean tryAcquirePermission() {
+            return true;
+        }
+
+        /**
+         * Does not throw an exception, because the CircuitBreaker is always closed in this state.
+         */
+        @Override
+        public void acquirePermission() {
+            // noOp
+        }
+
+        @Override
+        public void releasePermission() {
+            // noOp
+        }
+
+        @Override
+        public void onError(long duration, TimeUnit durationUnit, Throwable throwable) {
+            publishEventsIfPossible(circuitBreakerMetrics.onError(duration, durationUnit));
+        }
+
+        @Override
+        public void onSuccess(long duration, TimeUnit durationUnit) {
+            publishEventsIfPossible(circuitBreakerMetrics.onSuccess(duration, durationUnit));
+        }
+
+        private void publishEventsIfPossible(Result result) {
+            if (result.getComputeResult() != ABOVE_THRESHOLDS) {
+                return;
+            }
+
+            switch (result.getOutcome()) {
+                case ERROR:
```

Comment:
Before my change, there was no need to distinguish between slow call rate vs failure rate breaching the threshold. So `Result` just tells you if *ANY* of the thresholds were breached.

With the introduction of `METRICS_ONLY` mode, I'm emitting separate events when each of these thresholds are breached: `CircuitBreakerOnSlowCallRateExceededEvent` and `CircuitBreakerOnFailureCallRateExceededEvent`.

Does this answer your question?

Reply:
I think it would be sufficient to modify the existing enum `Result` and add `FAILURE_RATE_ABOVE_THRESHOLDS`, `SLOW_CALL_RATE_ABOVE_THRESHOLDS`. We could reuse `ABOVE_THRESHOLDS`  to signal thatn both thresholds have been exceeded.
But when we publish an event, the outcome of the last call (error, success, slow_error or slow_success) should be not important.

Reply:
Yes this is simpler. Thanks!